src/types.ts

```diff
@@ -33,10 +33,34 @@ export type {
 
 export type ImportType = ImportType_ | 'object' | 'type'
 
-export type NodeResolverOptions = {
+export type LegacyNodeResolverOptions = {
+  /** array of file extensions to search in order */
   extensions?: readonly string[]
-  moduleDirectory?: string[]
+  /** set to `false` to exclude node core modules (e.g.` fs`) from the search */
+  includeCoreModules?: boolean
+  /** directory (or directories) in which to recursively look for modules. Default "node_modules" */
+  moduleDirectory?: string
+  /** if true, doesn't resolve basedir to real path before resolving. This is the way Node resolves dependencies when executed with the --preserve-symlinks flag. Default to true */
+  preserveSymlinks?: boolean
+  /** Noop now, Previously a directory to begin resolving from */
+  basedir?: string
+  /** Noop now. Previously for require.paths array to, it is now noop */
   paths?: string[]
```

Comment:
Both the `basedir` option and the `paths` option are not supported by `unrs-resolver`. Is it possible to implement this, or should we just make it as noop?

This affects many unit tests IMHO.

Reply:
<img width="544" alt="image" src="https://github.com/user-attachments/assets/115932b3-69d5-46d9-8cd8-d56eabb485fe" />

I don't think `basedir` is exposed.

Reply:
IMO, we can mark `eslint-import-resolver-node` as optional peer dependency, if the user passed options we didn't support, we use original `eslint-import-resolver-node`, otherwise, use our internal node resolver instead.

Reply:
> IMO, we can mark `eslint-import-resolver-node` as optional peer dependency, if the user passed options we didn't support, we use original `eslint-import-resolver-node`, otherwise, use our internal node resolver instead.

The idea is that almost nobody uses `import/resolve`:

https://github.com/search?q=%2F%5B%27%22%5Dimport%5C%2Fresolve%5B%27%22%5D%2F&type=code

And only our unit tests use `import-x/resolve`:

https://github.com/search?q=%2F%5B%27%22%5Dimport-x%5C%2Fresolve%5B%27%22%5D%2F&type=code&p=2

So we can probably include this as a semi-breaking change and dropping `eslint-import-resovler-node` completely without a non-major version bump.

Reply:
> So we can probably include this as a semi-breaking change and drop eslint-import-resovler-node completely without a non-major version bump.

Well that's fine to me, but would marking `eslint-import-resolver-node` as optional peer requests additional efforts?

Reply:
> Well that's fine to me, but would marking `eslint-import-resolver-node` as optional peer requests additional efforts?

Well, we are doing ESM already. ESM doesn't have conditional require and will only early error. We can't do `import()` because it will be async (`synckit` won't help as the function is not transferable). We can still do `cjsRequire`, but personally I would prefer not to.

Also, npm installs optional peer deps by default (https://github.com/npm/feedback/discussions/225) so it is against our "removing `eslint-import-resolver-node` completely" effort.

Reply:
`cjsRequire` can be used in this case for `eslint-import-resolver-node`?

Reply:
IMHO, let's just remove it first, and probably no `eslint-plugin-import-x` users will be affected.

Implementing fallback will result in inconsistent behaviors (sometimes use `eslint-import-resolver-node`, sometimes use `unrs-resolver`), there might be this:

```js
// eslint.config.js
export default [
  { settings: { 'import-x/resolve': legacy } },
  { settings: { 'import-x/resolve': lessLegacy } }
];
```

Reply:
But we will still need to implement `paths`, otherwise the unit tests will fail.

Reply:
`modules` can not cover `paths` test cases? Then I'd like to still mark `resolver-node` as optional peer.

Reply:
> opts.paths - require.paths array to use if nothing is found on the normal node_modules recursive walk (probably don't use this)

I think `paths` behavior can be mocked when `found: false` and do another try through `modules: paths`?

Reply:
I don't get it. I just tried adding `paths` to `modules`. The tests are still failing.